---
title:
  # Title must be 150 characters or less
  "Visualizing the geography of genetic variants"
author:
  - name: Joseph H. Marcus
    affiliation: '1'
    # If corresponding author (only one corresponding author per manuscript)
    email: jhmarcus@uchicago.edu
  - name: John Novembre
    email: jnovembre@uchicago.edu
    affiliation: '1'
affiliation:
  - id: '1'
    department: Department of Human Genetics
    institution: University of Chicago
    city: Chicago
    state: IL
    country: USA
output: RmdTemplates::plos_article
csl: plos.csl
header-includes:
   - \usepackage{amsmath}
   - \usepackage{amsthm}
   - \usepackage{float}
bibliography: library.bib
---

```{r imports, echo=FALSE, message=FALSE, warning=FALSE}
library(png)
library(grid)
```

# Abstract {-}
One of the key characteristics of any genetic variant, beyond its potential phenotypic effects or its frequency, is its geographic distribution. The geographic distribution of a genetic variant can shed light on where the variant first arose, in what populations it spread within, and in turn can help one learn about historical patterns of migration, genetic drift, and natural selection. Collectively the geographic distribution of genetic variants can help to explain how populations have been related through time (e.g. levels of gene flow and divergence). For these reasons, visual inspection of geographic maps for genetic variants is common practice in genetic studies. Here we develop an interactive visualization tool for illuminating the geographic distribution of genetic variants. Through an efficient SQL database, REST API and dynamic front-end the [\textit{Geography of Genetic Variants (GGV)}](popgen.uchicago.edu/ggv) browser rapidly provides maps of allele frequencies in populations distributed across the globe.

# Introduction {-}
Genetics researchers often face the following problem: the researcher has identified one or many genetic variants of interest (e.g. from a genome-wide association, eQTL, or pharmacogenomic study) and would like to know the geographic distribution of the variant. For example, the researcher may hope to address: 1) implications for genomic medicine (e.g. is a risk allele geographically localized to a certain patient population?); 2) design of follow-up studies (e.g. what population should be studied to observe variant carriers?); or 3) the biology of the variant in question (e.g. does the variant correlate with a known environmental factor in a manner suggestive of some geographically localized selection pressure?)  @rosenberg2010genome, @novembre2009spatial, @coop2010using.  A simple geographic map of the distribution of a genetic variant can be incredibly insightful for these questions.

The geographic distribution of rare genetic variants (minor allele frequency $\leq .05$) can be particularly informative in determining an individual's ancestry. Rare variants tend to be localized in small geographic regions because they have arisen recently in human history and have had little time to spread across populations through migration events @nelson2012abundance, @mathieson2014demography. Surveying the collection of rare variants that an individual carries  can provide insight into the geographic origins of that individual's recent ancestors. One of us (J.N.) recently took advantage of the geographic localization of an individual's rare variants in helping to reaffirm the historically contested Native American ancestry of the Kennewick Man, an ancient individual from Washington State @KennewickManReport.

Contemporary population genetics researchers are also faced with the challenge of large, high dimensional datasets. For example, is not uncommon for a researcher in human genetics to have a dataset comprised of thousands of individuals measured at hundreds of thousands or even millions of single nucleotide variants (SNVs). One common approach to visualizing such high-dimensional data is to compress the SNV dimensions down to a small number of latent factors, using a method such as principal components analysis @price2006principal, @patterson2006population, or a model-based clustering method such as STRUCTURE @pritchard2000inference or ADMIXTURE @alexander2009fast. While these approaches are extremely valuable, researchers can use them to often without inspecting the underlying variant data in more detail. A natural approach to gaining more insight to the overall structure of a population genetic dataset is to visually inspect what geographic patterns arise in allele frequency maps. 

Unfortunately, generating geographic allele frequency maps is time-consuming for the average researcher as it requires a combination of data wrangling methods @furchedata and geographic plotting techniques that are unfamiliar to most. Our aim here is to produce a tailored system for rapidly constructing informative geographic maps of allele frequency variation. 

Our work is inspired by past tools such as the ALFRED database @rajeevan2011alfred and the maps available on the HGDP Selection browser @pickrell2009signals.  One of us (J.N.) developed the scripts for the HGDP Selection Browser maps using The Generic Mapping Tools (GMT) @wessel2013generic, a powerful system of geographic plotting scripts with a substantial learning curve.  The plots from the HGDP Selection Browser have proved useful, and have appeared in research articles e.g @pickrell2009signals, @coop2009role, books e.g. @dudley2013exploring, and have been ported and made available on the UCSC Genome Browser (available under the HGDP Allele Freq track of the browser).  

Reference datasets for population genetic variation has greatly expanded since the release of the HGDP Illumina 650Y dataset @li2008worldwide that formed the basis of the HGDP Selection Browser maps.  The most notable advance is the publication of the 1000 Genomes Phase 3 data @10002015global though additional datasets are coming online @meyer2012high, @lazaridis2014ancient.  

In addition, novel approaches for data visualization have become more widely available.  In particular, web-based visualization tools, such as Data Driven Documents (D3.js), offer powerful approaches to realize these aims through useful methods for interactivity, advantages of software development in modern web-browsers, a large open-source development community, and ease of sharing @bostock2011d3. 
   
Taking advantage of these recent advances, we aim to address significant visualization challenges that are inherit in the production of geographic allele frequency maps, including dynamic interaction, display of rare genetic variation, and representation of uncertainty in estimated allele frequencies due to variable sample sizes. 


```{r figure1, echo=FALSE, message=FALSE, fig.height=4, fig.width=4, fig.align='center', fig.cap='Schematic diagram of the Geography of Genetic Variants (GGV) browser. We compute minor allele frequencies from publicly available population genetic datasets from the 1000 Genomes Project, Human Genome Diversity Project (HGDP) and Population Reference Sample (POPRES). These allele frequencies are stored in a SQL database along side with geographic coordinates associated with each population. The front-end visualization was created using D3.js and accesses allele frequency and population meta-data from a REST API implemented in the python library flask.'}
img <- readPNG("figures/png/figure1.png")
grid.raster(img)
```

# Fundamental Approach {-}

The Geography of Genetic Variants browser (GGV) uses the SVG and mapping utilities of D3.js to generate interactive frequency maps, allowing for quick and dynamic displays of the geographic distribution of a genetic variant. 

In order to allow for easy access to large commonly used public genomic datasets, such as the 1000 Genomes project, Human Genome Diversity project and POPRES project, we have developed an SQL database and REST API for querying allele frequencies by chromosome and position or reference SNP identifier [Figure 1]. 

Users can query single variants by chromosome and position identifiers, by rsid @sherry2001dbsnp, or users can simply choose a random variant from within a dataset.  While many applications require inspection of the distribution of a specific variant, from our experience, it can be very helpful to quickly view the geographic distribution of several randomly chosen variants to quickly gain a sense of structure in a dataset. For instance, in data with deep population subdivision, it is obvious in the consistent patterns of differentiation observed across markers. We also expect this will be useful in teaching contexts – as it provides a highly visual way for learners to understand human genetic variation.

```{r figure2, echo=FALSE, message=FALSE, fig.height=4, fig.width=5, fig.align='center', fig.cap='Example map from the Geography of Genetic Variants browser. Each pie chart represents a population with the blue slice of the pie displaying the frequency of the global minor allele and the yellow slice of the pie displaying the frequency of the global major allele in each population.'}
img <- readPNG("figures/png/figure2.png")
grid.raster(img)
```

After a query, the GGV displays the frequency of a variant in a given population as a pie chart where each slice represents minor and major allele frequencies. Pie charts are displayed as points positioned at the population's region of origin where the projection / map-view is chosen based off of the geographic proximity of populations present in a given dataset [Figure 2]. 

# Representing variable certainty in frequency data {-}
One under-appreciated problem with allele frequency maps is that not all data points have equal levels of certainty. For some locations, sample sizes are small, and the reported allele frequency may be quite far from the true population frequency due to sampling error. To address this issue, we use varying transparency in a population's pie chart: sample allele frequencies with higher levels of sampling error will be made more transparent, and hence less visible on the map [Figure 3].   

```{r figure3, echo=FALSE, message=FALSE, fig.height=4, fig.width=5, fig.align='center', fig.cap='Example map from the Geography of Genetic Variants browser displaying the use of varying transparency of population pie charts to represent uncertainty in allele frequencies. The transparency is scaled in proportion to the number of observed chromosomes in each population for a particular variant.'}
img <- readPNG("figures/png/figure3.png")
grid.raster(img)
```

# Representing rare variants in frequency data {-}
An additional challenge is that allele frequencies between variants often differ greatly, sometimes by orders of magnitude in a single dataset. This has not been an pervasive problem until recently, as most population genetic samples were based on genotype arrays, biased towards variants that are common in human populations (5-50% in minor allele frequency). With the advent of next generation sequencing technologies and large samples of thousands of individuals, it is common for datasets to contain rare variants @10002015global, @nelson2012abundance, @tennessen2012evolution. 

```{r figure4, echo=FALSE, message=FALSE, fig.height=5, fig.width=5, fig.align='center', fig.cap='Example maps from the Geography of Genetic Variants browser displaying the use of frequency scales for more expressive representations of rare variation on geographic maps. The blue pie charts convey a give minor allele frequency out of 100 percent, the green out 1 percent, and the red out of .01 percent.'}
img <- readPNG("figures/png/figure4.png")
grid.raster(img)
```

In current visualization schemes, such as the HGDP Selection Browser, rare variants would be represented as narrow slivers in a pie chart, nearly invisible to the naked eye. To address this challenge we re-scale frequencies for rare variants, so that small frequencies become visible. Specifically, we will use a “frequency scale” that is indicated below the map, and redundantly using color, that will indicate a constant scale for the frequencies displayed [Figure 4]. Much like scientific notation, this allows a wide range of frequencies to be displayed].

# Interface features {-}
In many datasets where populations are sampled densely in geographic space, one problem is that allele frequency plots begin to overlap each other and obscure information. To address this issue, we use force-directed layouts of the populations such that no two points are overlapping each other, and yet the points will be pulled towards their true origins. Lines anchoring the points visibly to their original sampling locations will be used to make sure true sampling locations are indicated [Figure 5].

```{r figure5, echo=FALSE, message=FALSE, fig.height=6, fig.width=5, fig.align='center', fig.cap='Example maps from the Geography of Genetic Variants browser displaying the use of a force directed layout to limit visual clutter when many populations overlap in geographic position. The left map shows the original population locations while the right shows the application of the force directed layout.'}
img <- readPNG("figures/png/figure5.png")
grid.raster(img)
```

# Underlying Frequency API {-}
To interface population minor allele frequency data stored in the SQL database with the front-end D3.js visualization we use a REST API implemented in the python library Flask-RESTful @citeXXX. GET requests are submitted to the database via HTTP, returning json formatted allele frequency data and the meta-data associated with each population and genetic variant e.g. latitude, longitude, population label, sample size, and frequency scale. Genetic variants can be queried by chromosome position, rsid, or randomly. Example HTTP requests and json response can be seen below:

## Query by rsid {-}
```
http://popgen.uchicago.edu/ggv_api/freq_table?data="1000genomes_phase3_table"&rsID=rs1834640
[
  {
    "alleles": ["A", "G"], 
    "pos": ["-15.310139", "13.443182"], 
    "pop": "GWD", 
    "nobs": "226", 
    "xobs": "17", 
    "freqscale": 1, 
    "freq": [0.0752212389381, 0.9247787610619], 
    "chrom_pos": "15:48392165", 
    "rawfreq": 0.0752212389381
  },
  ...
]
```

## Query by chromosome position {-}

```
http://popgen.uchicago.edu/ggv_api/freq_table?data="1000genomes_phase3_table"&chr=14&pos=37690093
[
  {
    "alleles": ["G", "A"], 
    "pos": ["-15.310139", "13.443182"], 
    "pop": "GWD", 
    "nobs": "226", 
    "xobs": "0", 
    "freqscale": 0.01, 
    "freq": [0.0, 1.0], 
    "chrom_pos": "14:37690093", 
    "rawfreq": 0.0
  }
  ...
]
```

## Random query {-}

```
http://popgen.uchicago.edu/ggv_api/freq_table?data="1000genomes_phase3_table"&random_snp=True
[
  {
    "alleles": ["T", "C"], 
    "pos": ["-15.310139", "13.443182"], 
    "pop": "GWD", 
    "nobs": "226", 
    "xobs": "0", 
    "freqscale": 0.01, 
    "freq": [0.0, 1.0], 
    "chrom_pos": "5:42452893", 
    "rawfreq": 0.0
  }, 
  ...
]
```

# Caveats {-}
A major challenge of using a geographic representation of genetic variation in humans is that the samples must be associated with a geographic location.  While doing so is generally immensely helpful, it has inherent complexity.  For example,  practitioners must make choices regarding representing where an individual was sampled for the study (e.g. the city of a major research center) or choosing a location that is more representative of an individual's ancestral origins (e.g. as determined in practice by the birthplaces of recent ancestors, such as grandparents).  We do not proscribe a general solution to this problem and instead use a set of locations for each dataset that aligns with those used by the initial analysts of the data.  A future feature will be to allow alternative location schema to be used for the populations in a dataset. 


# Conclusion {-}
By allowing rapid generation of allele frequency maps, we hope to facilitate the interpretation of variant function and history by practicing geneticists.  We also hope the ability to query random variants from major human population genetic samples will allow students to appreciate the structure of human genetic diversity in a more approachable and intuitive form than alternative visualizations. 

# Acknowledgements {-}
Support for this work was provided by the NIH-BD2K initiative (1U01 CA198933-0).  The authors would also like to thank members of the Novembre Lab for supportive conversations. 

# References {-}

<!-- Has to be set for some wierd indentation issue -->
\setlength\parindent{0pt}

