\documentclass{bioinfo}
\copyrightyear{2015} \pubyear{2015}

\access{Advance Access Publication Date: Day Month Year}
\appnotes{Manuscript Category}

% testing atom git integration

\begin{document}
\firstpage{1}

\subtitle{Subject Section}

\title[short Title]{Visualizing the Geography of Genetic Variants}
\author[Sample \textit{et~al}.]{Joseph H. Marcus\,$^{\text{\sfb 1}}$ and John Novembre\,$^{\text{\sfb 1, 2,}*}$}
\address{$^{\text{\sf 1}}$Department of Human Genetics, University of Chicago, Chicago, 60637, US and \\
$^{\text{\sf 2}}$Department of Ecology and Evolution, University of Chicago, Chicago, 60637, US.}

\corresp{$^\ast$To whom correspondence should be addressed.}

\history{Received on XXXXX; revised on XXXXX; accepted on XXXXX}

\editor{Associate Editor: XXXXXXX}

\abstract{One of the key characteristics of any genetic variant is its geographic distribution.  The geographic distribution can shed light on where an allele first arose, what populations it has spread to, and in turn on how migration, genetic drift, and natural selection have acted. The geographic distribution of a genetic variant can also be of great utility for medical/clinical geneticists and collectively many genetic variants can reveal population structure. Here we develop an interactive visualization tool for rapidly displaying the geographic distribution of genetic variants. Through a REST API and dynamic front-end the \href{popgen.uchicago.edu/ggv}{\textit{Geography of Genetic Variants (GGV)}} browser provides maps of allele frequencies in populations distributed across the globe.\\
\textbf{Contact:} \href{jhmarcus@uchicago.edu}{jhmarcus@uchicago.edu}, \href{jnovembre@uchicago.edu}{jnovembre@uchicago.edu}\\
\textbf{Supplementary information:} Supplementary data are available at \textit{Bioinformatics}
online.}

\maketitle

\section{Introduction}

Genetics researchers often face the problem that they have identified one or many genetic variants of interest using an approach such as a genome-wide association study and then would like to know the geographic distribution of the variant. For example, the researcher may hope to address: 1) implications for genomic medicine \citep[e.g. Is a risk allele geographically localized to a certain patient population? What population should be studied to observe variant carriers?][]{rosenberg2010genome}; or 2) the evolutionary history of the variant in question \citep[e.g. does the variant correlate with a known environmental factor in a manner suggestive of some geographically localized selection pressure?][]{novembre2009spatial}.  A simple geographic map of the distribution of a genetic variant can be incredibly insightful for these questions.

Contemporary population genetics researchers are also faced with the challenge of large, high-dimensional datasets. For example, it is not uncommon for a researcher in human genetics to have a dataset comprised of thousands of individuals measured at hundreds of thousands or even millions of single nucleotide variants (SNVs). One common approach to visualizing such high-dimensional data is to compress the SNV dimensions down to a small number of latent factors, using a method such as principal components analysis \citep{patterson2006population}, or a model-based clustering method such as STRUCTURE \citep{pritchard2000inference}. While these methods are extremely valuable, researchers can use them too often without inspecting the underlying variant data in more detail. A natural approach to gaining more insight to the overall structure of a population genetic dataset is to visually inspect what geographic patterns arise in allele frequency maps.

Unfortunately, generating geographic allele frequency maps is time-consuming for the average researcher as it requires a combination of data-wrangling methods \citep{kandel2011research} and map-making techniques that are unfamiliar to most. Our aim here is to produce a tailored system for rapidly constructing informative geographic maps of allele frequency variation.

Our work is inspired by past tools such as the ALFRED database \citep{rajeevan2011alfred} and the maps available on the HGDP Selection browser \citep{pickrell2009signals} whose allele frequency output and plots have been used in research articles \citep[e.g.][]{pickrell2009signals,coop2009role}, books \citep[e.g.][]{dudley2013exploring}, and have been made available on the UCSC Genome Browser \citep[available under the HGDP Allele Freq track of the browser,][]{kent2002human}.

Taking advantage of recent advances in web-based visualization tools \citep{bostock2011d3}, we aim to address the significant visualization challenges that are inherit in the production of geographic allele frequency maps for large population genomic datasets, including dynamic interaction, display of rare genetic variation, and representation of uncertainty in estimated allele frequencies due to variable sample sizes.

\section{Approach}

\begin{figure}[!tpb]%figure1
\centerline{\includegraphics[scale=0.22]{figure1.jpg}}
\caption{Example screenshot from the Geography of Genetic Variants browser using \citet{10002015global} data. Each pie chart represents a population with the blue slice of the pie displaying the frequency of the global minor allele.}\label{fig:01}\label{fig:screenshot}
\end{figure}

The Geography of Genetic Variants browser (GGV) uses the scalable vector graphics and mapping utilities of D3.js \citep{bostock2011d3}. The front-end provides legends for the map and various configuration boxes to allow users to query different datasets or choose visualization options.

In order to allow for easy access to commonly used public genomic datasets, such as the 1000 Genomes project \citep{10002015global} or Human Genome Diversity project \citep{li2008worldwide}, we have developed a REST API \citep{grinberg2014flask}.  The API allows retreival of SNVs by position, rsid \citep{sherry2001dbsnp} or at random. After a query, the GGV displays the allele frequencies as a collection of pie charts where each represents the frequency of the globally minor allele in a single population [Figure \ref{fig:screenshot}].

We implement a variety of features: \textbf{(1) Rare variants.} Many alleles are rare \citep[e.g.][]{10002015global}, and displaying them can be a challenge with proportional scales that range from zero to one. To address this challenge we re-scale frequencies , so that small frequencies become visible. Specifically, we  use a frequency scale that is indicated in a legend below the map and represented by varying color in the pie charts [Figure \ref{fig:screenshot}, Supplementary Figure S1,   Supplementary Table S1]. \textbf{(2) Uncertainty in frequency data.} We use varying transparency in a population's pie chart: estimated frequencies with higher levels of sampling error (e.g.~those from samples with $n < 30$) are made more transparent, and hence less visible, on the map [Figure \ref{fig:screenshot}, Supplementary Figure S2]. \textbf{(3) Overlapping populations.} We use force-directed layouts of the populations such that no two points are overlapping each other, and yet the points will be pulled towards their true origins [Figure \ref{fig:screenshot}, Supplementary Figure S3].  Also, by hovering the mouse cursor over any population, a user can see the population labels and precise frequency information.

By allowing rapid generation of allele frequency maps, we hope to facilitate the interpretation of variant function and history by practicing geneticists. Also, for students of human diversity, it is often difficult to conceptualize classic statements regarding how most variation in humans is shared among populations \citep{lewontin1972apportionment} and how the fixation index $F_{ST}$ is relatively low globally \citep[10-15\%][]{10002015global}.  We hope that the ability to query random variants from major human population genetic samples will allow students to appreciate the structure of human genetic diversity in an approachable and intuitive form.

\section*{Acknowledgements}

We acknowledge the Research Computer Center at the University of Chicago, especially Jeff Tharsen and Alex Mueller, for on-going support and development, as well as John Zekos for server administration support and members of the Novembre Lab. Support for this work was provided by the National Institutes of Health via the Big Data to Knowledge initiative (1U01 CA198933-0, JN) and the National Institute Of General Medical Sciences under training grant award number T32GM007197 (JHM).

\bibliographystyle{natbib}
%\bibliographystyle{achemnat}
%\bibliographystyle{plainnat}
%\bibliographystyle{abbrv}
%\bibliographystyle{bioinformatics}
%\bibliographystyle{plain}

\bibliography{main}

\end{document}
